doc/dev/crimson/poseidonstore.rst

```diff
@@ -0,0 +1,470 @@
+Also, each reactor thread has a static set of SPs.
+
+
+Cache
+-----
+There are mainly two cache data structures; onode cache and block cache.
+It looks like below.
+
+1.Onode cache:
+lru_map <OID, OnodeRef>;
+Onode {
+  extent tree block_maps;
+  omap lists omaps;
+}
+2. Block cache:
+Omap cache --> lru_map <OID, <omap_key, value>>
+Data cache --> lru_map <OID, <extent, value>>
+
+To fill the onode data structure, the target onode needs to be retrieved using the prefix tree.
+Block cache is used for caching a block contents. For a transaction, all the updates to blocks 
+(including object meta block, data block) are first performed in the in-memory block cache.
+After writing a transaction to the WAL, the dirty blocks are flushed to their respective locations in the 
+respective partitions.
+PoseidonStore can configure cache size for each type. Simple LRU cache eviction strategy can be used for both.
+
+
+Sharded partitions (with cross-SP transaction)
+--------------------------------------------
+The entire disk space is divided into a number of chunks called sharded partitions (SP).
+The prefixes of the parent collection ID (original collection ID before collection splitting. That is, hobject.hash) 
+is used to map any collections to SPs.
+We can use BlueStore's approach for collection splitting, changing the number of significant bits for the collection prefixes.
+Because the prefixes of the parent collection ID do not change even after collection splitting, the mapping between 
+the collection and SP are maintained.
+The number of SPs may be configured to match the number of CPUs allocated for each disk so that each SP can hold 
+a number of objects large enough for cross-SP transaction not to occur.
+
+In case of need of cross-SP transaction, we could use the global WAL. The coordinator thread (mainly manages global partition) handles 
+cross-SP transaction via acquire the source SP and target SP locks before processing the cross-SP transaction. 
+Source and target probably are blocked.
+
+For the load unbalanced situation, 
+Poseidonstore can create partitions to make full use of entire space efficiently and provide load balaning.
+
+
+COW/Clone
+---------
+As for CoW/Clone, a clone has its own onode like other normal objects.
+
+Although each clone has its own onode, data blocks should be shared between the original object and clones
+if there are no changes on them to minimize the space overhead.
+To do so, the reference count for the data blocks is needed to manage those shared data blocks.
+
+To deal with the data blocks which has the reference count, poseidon store makes use of shared_blob
+which maintains the referenced data block.
+
+As shown the figure as below, by using shared_blob tracked by shared_blob_list in the superblock, 
+the blocks for the clone are managed using shared_blob with a reference count.
+
+.. ditaa::
+
+
+    +----------+         +---------+
+    | Object A |----- |--| Clone A |---------
```

Comment:
This diagram is pretty confusing.  There is no distinction between objects and clones, so I think you mean object A and object B.  Second, where is the refcount information for the shared_block stored relative to object A's onode, object B's onode and the actual data blocks?  I think that would be the most important thing to convey.